### PR TITLE
Add normalized exponential transform

### PR DESCRIPTION
**Pull request by sethaxen**
This PR adds the "Normalized Exponential" transform (name TBD). It's the special case of the "Normalized Gamma" (also TBD) proposed by @nsiccha on Slack for $\alpha=\mathbf{1}$. To support Normalized Gamma for arbitrary $\alpha$, Stan needs an implementation of the inverse incomplete gamma function.

### CODE REVIEW

File: transforms/simplex/NormalizedExponential.stan [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    vector[N] z;
+    real log_u;
+    for (i in 1:N) {
+      log_u = std_normal_lcdf(y[i]);
+      z[i] = exponential_log_qf(log_u);
+      r += z[i];
+      log_det_jacobian += std_normal_lpdf(y[i]);
+    }
+    for (i in 1:N)
+      x[i] = z[i] / r;
```
Comment (spinkney) [inline transforms/simplex/NormalizedExponential.stan:27]: the loop can be removed and we can just assign  `x = z / r;`

Comment (sethaxen) [inline transforms/simplex/NormalizedExponential.stan:27]:
```suggestion
    x = z / r;
```

**Comment by sethaxen** [comment]
For normalized gamma, I think the relevant issue to track is https://github.com/stan-dev/math/issues/2509.

**Comment by nsiccha** [comment]
@sethaxen I just realized that `exponential_log_qf(std_normal_lcdf(x)) = -log1m_exp(std_normal_lcdf(x)) = -std_normal_lccdf(x)`, though I'm not sure how stable Stan's implementation is, see https://github.com/stan-dev/math/issues/1985

**Comment by sethaxen** [comment]
Indeed, it seems that not as much work has been put into stability of that function. I think what we're doing here is probably preferable for now.

**Comment by nsiccha** [comment]
Yeah, I had put this here mainly so that I don't forget about it :) (But forgot to mention that :sweat_smile:)

**Pull request merged**